util/patch/utils.go

```diff
@@ -92,18 +98,80 @@ func unsafeUnstructuredCopy(obj *unstructured.Unstructured, focus patchType, isC
 			res.Object[key] = value
 		}
 
-		// If we've determined that we're able to interface with conditions.Setter interface,
-		// when dealing with the status patch, remove the status.conditions sub-field from the object.
-		if isConditionsSetter && focus == statusPatch {
-			// NOTE: Removing status.conditions changes the incoming object! This is safe because the condition patch
-			// doesn't use the unstructured fields, and it runs before any other patch.
-			//
-			// If we want to be 100% safe, we could make a copy of the incoming object before modifying it, although
-			// copies have a high cpu and high memory usage, therefore we intentionally choose to avoid extra copies
-			// given that the ordering of operations and safety is handled internally by the patch helper.
-			unstructured.RemoveNestedField(res.Object, "status", "conditions")
+		// If we've determined that new or old condition must be set,
+		// when dealing with the status patch, remove corresponding sub-fields from the object.
+		// NOTE: Removing conditions sub-fields changes the incoming object! This is safe because the condition patch
+		// doesn't use the unstructured fields, and it runs before any other patch.
+		//
+		// If we want to be 100% safe, we could make a copy of the incoming object before modifying it, although
+		// copies have a high cpu and high memory usage, therefore we intentionally choose to avoid extra copies
+		// given that the ordering of operations and safety is handled internally by the patch helper.
+
+		if len(metav1ConditionsFields) > 0 {
+			unstructured.RemoveNestedField(res.Object, metav1ConditionsFields...)
+		}
+
+		if len(clusterv1ConditionsFields) > 0 {
+			unstructured.RemoveNestedField(res.Object, clusterv1ConditionsFields...)
 		}
 	}
 
 	return res
 }
+
+var (
+	clusterv1ConditionsType = reflect.TypeOf(clusterv1.Conditions{})
+	metav1ConditionsType    = reflect.TypeOf([]metav1.Condition{})
+)
+
+func identifySupportedConditions(obj runtime.Object) ([]string, []string, error) {
```

Comment:
Do we need some warning somewhere that patching conditions with the patchHelper doesn't work with Unstructured? (or error out if we can detect it)

Or different angle. what happens today if I try to use the patch helper with Unstructured / what changes with this PR? (maybe not that much as Unstructured never implemented Getter/Setter)